hcl2template/types.build.go

```diff
@@ -211,5 +211,30 @@ func (p *Parser) decodeBuildConfig(block *hcl.Block, cfg *PackerConfig) (*BuildB
 		}
 	}
 
+	// Creates a bucket if either a hcp_packer_registry block is set or PAR is enabled via
+	// environment variable
+	if build.HCPPackerRegistry != nil || env.IsPAREnabled() {
+		var err error
+		cfg.bucket, err = packerregistry.NewBucketWithIteration(packerregistry.IterationOptions{
+			TemplateBaseDir: cfg.Basedir,
+		})
+		if err != nil {
+			diags = append(diags, &hcl.Diagnostic{
+				Summary:  "Unable to create a valid bucket object for HCP Packer Registry",
+				Detail:   fmt.Sprintf("%s", err),
+				Severity: hcl.DiagError,
+			})
+		}
+		cfg.bucket.LoadDefaultSettingsFromEnv()
+		if cfg.bucket.Slug == "" && build.Name != "" {
+			cfg.bucket.Slug = build.Name
+		}
+		build.HCPPackerRegistry.WriteBucketConfig(cfg.bucket)
+
+		for _, source := range build.Sources {
+			cfg.bucket.RegisterBuildForComponent(source.String())
+		}
+	}
```

Comment:
Is this doing any network calls ? We are decoding the config here, I think that this should be called afterwards; like on initialization; this way I can run `packer validate` on a config file with a blank HCP env.

Reply:
This doesn't do any network calls. It just initializes things. I tested with packer validate to make sure it wasn't doing such a thing.

Reply:
No network calls being done here. It is solely adding the names of each source to a list of potential builds for the iteration. All network calls pertaining to the registry happen right before we start building to initialize the bucket and iteration. 

That aside for the packer validate if a block is set does it validate actually check for HCP credentials? Great call out @azr

Reply:
The credentials validation is not there yet! I can add it somewhere over here 🤔

Reply:
nice !

Reply:
For the record, I don't think validate should call out to HCP or check for HCP creds. I see validate as a way to check that the config has all the required configuration in place. No that the environment is setup to build. 

I can run `packer validate` on an Amazon config and it does not fail if my creds are not set. I think we should behave the same way.  Thoughts?

Reply:
Yeah, since there was no other validation, I thought this was doing it too !

Reply:
Yes, I think we all agree, Wilken ! I think Moss wants to run HCP (networked) validation before building (so that would be not called on `validate` I think)

Reply:
Nice! I'm leaving the way it is! The HCP creds only checked when the client is actually created for use (upon packer build)